sssd_test_framework/utils/socket.py

```diff
@@ -0,0 +1,245 @@
+        """
+        Check if the responder's socket unit is active.
+
+        :return: True if active, False otherwise.
+        :rtype: bool
+        """
+        result = self.host.conn.exec(["systemctl", "is-active", self.socket_unit], raise_on_error=False)
+        return result.stdout.strip() == "active"
+
+    def is_service_active(self) -> bool:
+        """
+        Check if the responder's service unit is active.
+
+        :return: True if active, False otherwise.
+        :rtype: bool
+        """
+        result = self.host.conn.exec(["systemctl", "is-active", self.service_unit], raise_on_error=False)
+        return result.stdout.strip() == "active"
+
+    def get_socket_path(self) -> Optional[str]:
+        """
+        Get the UNIX socket path for the responder.
+
+        :return: Socket path if found, None otherwise.
+        :rtype: Optional[str]
+        """
+        result = self.host.conn.exec(
+            ["systemctl", "show", self.socket_unit, "--property=Listen"], raise_on_error=False
+        )
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to get socket path for {self.socket_unit}: {result.stderr}")
+
+        match = re.search(r"Listen=([^\s]+)", result.stdout)
+        return match.group(1) if match else None
+
+    def trigger_socket(self, command: str, args: list[str] = []) -> ProcessResult:
+        """
+        Trigger the responder by running a command that activates the socket.
+
+        :param command: Command to trigger the responder (e.g., 'getent', 'sssctl').
+        :type command: str
+        :param args: Arguments for the command.
+        :type args: list[str]
+        :return: Process result of the command.
+        :rtype: ProcessResult
+        """
+        result = self.host.conn.exec([command, *args], raise_on_error=False)
+        if result.rc != 0 and "Connection refused" in result.stderr:
+            raise RuntimeError(f"Socket activation failed for {self.responder}: {result.stderr}")
+        return result
+
+
+class SSSDSocketUtils(MultihostUtility[MultihostHost]):
+    """
+    Utility class to manage SSSD socket-activated responders in tests.
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        Initialize the socket utilities for SSSD responders.
+
+        :param host: Remote host.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+        self.responders: dict[str, SSSDSocketResponder] = {}
+
+    def get_responder(self, responder: str) -> SSSDSocketResponder:
+        """
+        Get or create an SSSDSocketResponder object for the given responder.
+
+        :param responder: Responder type (e.g., 'nss', 'pam').
+        :type responder: str
+        :return: Responder object.
+        :rtype: SSSDSocketResponder
+        """
+        if responder not in self.responders:
+            self.responders[responder] = SSSDSocketResponder(self.host, responder)
+        return self.responders[responder]
```

Comment:
This looks like an overkill, simple iterate the responders here as properties - they can even be lazy if you want, (@cachedproperty) but that's probably not needed. Then you can remove all the checking code.

Reply:
removed